xdsl/ir.py

```diff
@@ -144,9 +144,27 @@ def name(self) -> str | None:
         return self._name
 
     @name.setter
-    def name(self, name: str):
-        if self._name_regex.fullmatch(name):
+    def name(self, name: str | None):
+        if name is None:
+            self._name = None
+        # emit a warning if the string is a purely numeric value
+        elif name.isnumeric():
```

Comment:
A wise man once told me, that `isnumeric` is completely broken. (like `'⅓'.isnumeric() == True`)

Reply:
You are 100% correct, this is much cleaner now!